### PR TITLE
Use create instead of save

### PR DESCRIPTION
**Pull request by siemensikkema**
This allows specifying an id when doing saveOrUpdate().
To make this clear I've deprecated saveOrUpdate and introduced createOrUpdate which saveOrUpdate forwards to.

### CODE REVIEW

File: Sources/Sugar/Helpers/Model.swift [hunk 1]
```diff
@@ -11,7 +11,22 @@ public extension Model {
...
 
-    func saveOrUpdate (
+    @available(*, deprecated, message: "use `creatOrUpdate(given:withSoftDeleted:restore:on)`")
+    func saveOrUpdate(
+        given filters: [FilterOperator<Self.Database, Self>],
+        withSoftDeleted: Bool = false,
+        restore: Bool = false,
+        on db: DatabaseConnectable
+    ) throws -> Future<Self> {
+        return try createOrUpdate(
```
Comment (emrahsifoglu) [inline Sources/Sugar/Helpers/Model.swift:21]: **return** might be removed.

**Review by heidipuk — APPROVED** [review]
lgtm

File: Sources/Sugar/Helpers/Model.swift (hunk 1 above)
Comment (siemensikkema) [reply Sources/Sugar/Helpers/Model.swift:21]: Would be a breaking change for project depending on this that don't run swift 5.1

**Comment by emrahsifoglu** [comment]
Are we moving away from **codebeat** to **codecov**?

**Pull request merged**